### PR TITLE
core: fix golint nits

### PR DESCRIPTION
**Pull request by tooooolong**
Hi guys:

I'm learning the code of go-ethereum.
And I use `golint` to lint the code, then found some mistakes.

### What changes

1. fix typos
2. methods recevier of struct should be same and should not use `self` keywords
3. comments improve

There are many warning of comments, can you guys help to improve the comments.

most of warning is like this
```
core/state/state_object.go:33:6: exported type Code should have comment or be unexported
core/state/state_object.go:35:1: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self"
```

### How to represent
```
golint ./... | grep -v vendor
```

### CODE REVIEW

**Comment by GitCop** [comment]
Thank you for your contribution! Your commits seem to not adhere to the repository coding standards

 * Commit: bf96a7ae5ee58b86a863f787b13705e59d745b12
  * Commits must be prefixed with the package(s) they modify

Please check the [contribution guidelines](https://github.com/ethereum/go-ethereum#contribution) for more details.

---
This message was auto-generated by https://gitcop.com

**Comment by obscuren** [comment]
@tooooolong thank you!

@fjl Are you reviewing this?

**Comment by obscuren** [comment]
Fixed a merge conflict. Please squash it out when merging.

**Comment by tooooolong** [comment]
If you are glad to see the code refactoring like this, i will fix the rest warnings report from golint.

File: core/blockchain.go [hunk 1]
```diff
@@ -1168,21 +1169,21 @@ func (self *BlockChain) reorg(oldBlock, newBlock *types.Block) error {
 		log.Error("Impossible reorg, please file an issue", "oldnum", oldBlock.Number(), "oldhash", oldBlock.Hash(), "newnum", newBlock.Number(), "newhash", newBlock.Hash())
 	}
 	var addedTxs types.Transactions
-	// insert blocks. Order does not matter. Last block will be written in ImportChain itself which creates the new head properly
+	// insert blocks. Order does not matter. Last block will be written in ImportChain itbc which creates the new head properly
```
Comment (karalabe) [inline core/blockchain.go]: "itbc" :)

File: core/blockchain.go [hunk 2]
```diff
@@ -478,41 +478,41 @@ func (bc *BlockChain) insert(block *types.Block) {
 	}
 }
 
-// Accessors
+// Genesis Accessors
```
Comment (karalabe) [inline core/blockchain.go]: This was supposed to be a more general code "section" comment. Bad idea tbh. Might as well convert it into a proper doc for this single method. Please change to:

`Genesis retrieves the chain's genesis block.`

File: core/blockchain.go [hunk 3]
```diff
@@ -56,7 +56,7 @@ const (
 	blockCacheLimit     = 256
 	maxFutureBlocks     = 256
 	maxTimeFutureBlocks = 30
-	// must be bumped when consensus algorithm is changed, this forces the upgradedb
+	// BlockChainVersion must be bumped when consensus algorithm is changed, this forces the upgradedb
```
Comment (karalabe) [inline core/blockchain.go]: This isn't really a doc for `BlockChainVersion`, rather just a memo. Alas it's wrong and `upgradedb` is gone a long time ago, so we should rewrite to:

`BlockChainVersion ensures that an incompatible database forces a resync from scratch.`

File: core/tx_pool.go [hunk 4]
```diff
@@ -209,6 +209,7 @@ func (pool *TxPool) resetState() {
 	pool.promoteExecutables(currentState)
 }
 
+// Stop stops a TxPool
```
Comment (karalabe) [inline core/tx_pool.go]: Please add a period to the end of the comment, and write out `transaction pool`, not `TxPool`.

File: core/tx_pool.go [hunk 5]
```diff
@@ -238,6 +239,7 @@ func (pool *TxPool) SetGasPrice(price *big.Int) {
 	log.Info("Transaction pool price threshold updated", "price", price)
 }
 
+// State returns the state of TxPool
```
Comment (karalabe) [inline core/tx_pool.go]: Please add a period to the end of the comment, and write out `transaction pool`, not `TxPool`.

**Comment by karalabe** [comment]
I've cherry picked this PR into a new one on top of master and fixed the issues I've raised. Merging when green. https://github.com/ethereum/go-ethereum/pull/14515

**Comment by tooooolong** [comment]
@karalabe  Hi, I have lots work to do those days, and i have planed to get those work done this weekend.  Sorry about that.

**Comment by karalabe** [comment]
No worries, thank you very much for the effort. I'm just trying to wrap up the open commits for next week's release and wanted to get everything in before the weekend.

**Comment by tooooolong** [comment]
thank you very much  ❤️

**Pull request closed without merging**